deeppavlov/models/kbqa/query_generator.py

```diff
@@ -0,0 +1,597 @@
+
+        if not candidate_outputs and entities_from_ner:
+            entity_ids = self.get_entity_ids(entities_from_ner)
+            log.debug(f"entities_from_ner {entities_from_ner}")
+            log.debug(f"entity_ids {entity_ids}")
+            self.template_num = int(template_type[0])
+            candidate_outputs = self.find_candidate_answers(question, entity_ids, rels_from_template=None)
+
+        if self.return_answers:
+            answer = self.rel_ranker([question], candidate_outputs, True)
+            return [answer]
+        else:
+            return candidate_outputs
+
+    def get_entity_ids(self, entities: List[str]) -> List[List[str]]:
+        entity_ids = []
+        for entity in entities:
+            entity_id, confidences = self.linker(entity)
+            entity_ids.append(entity_id[:15])
+        return entity_ids
+
+    def find_candidate_answers(self, question: str,
+                               entity_ids: List[List[str]],
+                               rels_from_template: List[Tuple[str]]) -> List[Tuple[str]]:
+        candidate_outputs = []
+
+        if self.template_num == 0 or self.template_num == 1:
+            candidate_outputs = self.complex_question_with_number_solver(question, entity_ids)
+            if not candidate_outputs:
+                self.template_num = 7
+
+        if self.template_num == 2 or self.template_num == 3:
+            candidate_outputs = self.complex_question_with_qualifier_solver(question, entity_ids)
+
+        if self.template_num == 4:
+            candidate_outputs = self.questions_with_count_solver(question, entity_ids)
+
+        if self.template_num == 5:
+            candidate_outputs = self.maxmin_one_entity_solver(question, entity_ids[0][:self.entities_to_leave])
+
+        if self.template_num == 6:
+            candidate_outputs = self.maxmin_two_entities_solver(question, entity_ids)
+            if not candidate_outputs:
+                self.template_num = 5
+                candidate_outputs = self.maxmin_one_entity_solver(question, entity_ids[0][:self.entities_to_leave])
+
+        if self.template_num == 7:
+            candidate_outputs = self.two_hop_solver(question, entity_ids, rels_from_template)
+
+        if self.debug:
+            log.debug("candidate_rels_and_answers:\n" + '\n'.join([str(output) for output in candidate_outputs]))
+
+        return candidate_outputs
+
+    def complex_question_with_number_solver(self, question: str, entity_ids: List[List[str]]) -> List[Tuple[str]]:
+        question_tokens = nltk.word_tokenize(question)
+        ex_rels = []
+        for entity in entity_ids[0][:self.entities_to_leave]:
+            ex_rels += self.wiki_parser("rels", "forw", entity, type_of_rel="direct")
+        ex_rels = list(set(ex_rels))
+        scores = self.rel_ranker([question], ex_rels, False) if self.return_answers else self.rel_ranker(question, ex_rels)
```

Comment:
`RelRankerInfer.__call__` method receives only two arguments, `False` is unexpected.